### PR TITLE
[Newsletter] Create default membership product silently in newsletter context

### PR DESCRIPTION
**Pull request by millerf**
Fixes https://github.com/Automattic/gold/issues/49

## Proposed changes:

* Add a separate selector for the Newsletter products that resolves and silences the creation of the default product  

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Create a new site
* Sandbox the site
* Go to "new post" page and open the newsletter 
* **You should not see** any snackbar saying "Successfully created payment plan"
Screenshot 2023-05-25 at 15 34 07

* Jest tests:
```
cd projects/plugins/jetpack && pnpm test-extensions extensions/store/membership-products/test
```
I tried to implement some resolver tests, but I failed to make them work as expected....

### CODE REVIEW

File: projects/plugins/jetpack/extensions/store/membership-products/actions.js [hunk 1]
```diff
@@ -32,7 +32,8 @@ export const saveProduct =
 		product,
 		productType = PRODUCT_TYPE_PAYMENT_PLAN,
 		setSelectedProductId = () => {},
-		callback = () => {}
+		callback = () => {},
+		silentProductCreation = false
```
Comment (romarioraffington) [inline projects/plugins/jetpack/extensions/store/membership-products/actions.js]: What do you think about giving this a more descriptive name like `shouldDisplayProductCreationNotice`?

Same feedback for the vars and test cases in the other files as well

Comment (millerf) [reply projects/plugins/jetpack/extensions/store/membership-products/actions.js]: Good idea.Fixed in 2f6d1d26d6

File: projects/plugins/jetpack/extensions/store/membership-products/resolvers.js [hunk 2]
```diff
@@ -81,11 +87,20 @@ const setDefaultProductIfNeeded = ( selectedProductId, setSelectedProductId, sel
 	}
 };
 
+export const getNewsletterProducts = (
+	productType = PRODUCT_TYPE_PAYMENT_PLAN,
+	selectedProductId = 0,
+	setSelectedProductId = () => {}
+) =>
+	// Returns the products, but silences the snack bar if a default product is created
+	getProducts( productType, selectedProductId, setSelectedProductId, false );
```
Comment (romarioraffington) [inline projects/plugins/jetpack/extensions/store/membership-products/resolvers.js:97]: Optional feedback here. Personally, I find params either to read verses having to look up the fn definition

```suggestion
       $shouldDisplayProductCreationNotice = false;
	getProducts( productType, selectedProductId, setSelectedProductId, shouldDisplayProductCreationNotice );
```

**Review by romarioraffington — DISMISSED** [review]

**Comment by romarioraffington** [comment]
Added an optional feedback. It also appears that a required test failing

**Comment by millerf** [comment]
@jeherve I am unsure if those tests are failing because of this PR or if this is the "usual" tests failing...

**Comment by jeherve** [comment]
> I am unsure if those tests are failing because of this PR or if this is the "usual" tests failing...

I can't tell since they're running at the moment, but if the tests are required, then they should not be flaky. If they're failing on your PR and if the failure is not related to your PR, then you would see the same test failing on all other recent PRs in the repo.
Let me know if they fail again after this new run!

**Comment by millerf** [comment]
@jeherve YEs, they are still failing. I also tested on trunk: https://github.com/Automattic/jetpack/pull/31138

**Comment by jeherve** [comment]
That "E2E Tests / Jetpack blocks e2e tests" check isn't required, so you should be able to merge even when they're failing.

They're a bit flaky, which is why we retry them when they run for the mirror copy of `trunk`. You can see the recent runs here:
https://a8c-jetpack-e2e-reports.s3.amazonaws.com/reports/jetpack-production/report/index.html

File: projects/plugins/jetpack/extensions/store/membership-products/resolvers.js (hunk 2 above)
Comment (millerf) [reply projects/plugins/jetpack/extensions/store/membership-products/resolvers.js:97]: I personnally find it too noisy, when one can just look at the definition of `getProducts()` with one click.  But thanks for the suggestion.

Comment (millerf) [reply projects/plugins/jetpack/extensions/store/membership-products/resolvers.js:97]: I personnally find it too noisy, when one can just look at the definition of `getProducts()` with one click.  But thanks for the suggestion.

**Review by jeherve — COMMENTED** [review]
I'm seeing this when testing this on a new WordPress.com simple site, after connecting my site to Stripe, when I'm landing back in the editor:

Screenshot 2023-06-02 at 12 45 43
Screenshot 2023-06-02 at 12 46 38

**Comment by millerf** [comment]
@jeherve Nice catch, I am still wondering why I didn't see it before... 🤔 
I have fixed it (I missed an optional parameter)

https://github.com/Automattic/jetpack/assets/790558/09821c52-8352-41e2-b397-bc4de46ca6b6

**Comment by anomiex** [comment]
I have to ask, what's going on with all these Friday PRs you're flagging for cherry-picking? Why did these not get done earlier in the week, or even better last week before the previously-announced code freeze? When I click through to the linked bugs they were known about back then.

At this point this stuff is going to get basically no testing in the beta before being released.

**Comment by millerf** [comment]
> Why did these not get done earlier in the week, or even better last week before the previously-announced code freeze?

Because most of those were either discovered during beta tests, or were last-minute decisions.

**Comment by anomiex** [comment]
> Because most of those were either discovered during beta tests, or were last-minute decisions.

This PR itself was created on May 25, but it didn't get reviewed until May 30 and still isn't merged. If something is created that late in the month and is wanted in the release, getting it reviewed and merged should have been a higher priority.

The others I've looked at seemed similar to me: an issue was identified fairly early on, but fixing it seemingly wasn't treated as a priority until the end of beta week.

**Comment by millerf** [comment]
> This PR itself was created on May 25,

Yes, that is what I call beta test phase. 

And as you have seen, there were a couple of those PRs lying around. The ones you haven't seen, as were fixed before, had a bigger priority.

What I want to convey is that we have pushed the higher priority first, and the lesser (as this one) for after. 

Now, if this one cannot be cherry-picked, that is fine.

**Comment by anomiex** [comment]
> Now, if this one cannot be cherry-picked, that is fine.

At this point I think that will be for the best. Thanks!

**Comment by millerf** [comment]
@jeherve would you mind having a look at this? It has been stalled for a while... I just rebased.

**Review by jeherve — APPROVED** [review]
I believe that works. I've had some issues testing the payment flow behavior on WordPress.com Simple though, so I'd appreciate if someone from Gold could test too!

**Comment by romarioraffington** [comment]
This is already merged but LGTM for the Gold side

**Pull request merged**